src/android/SimpleCameraPreview.java

```diff
@@ -262,21 +265,36 @@ private boolean capture(boolean useFlash, CallbackContext callbackContext) {
         return true;
     }
 
-    private boolean torchSwitch(boolean torchState, CallbackContext callbackContext) {
-      if (fragment == null) {
-        callbackContext.error("Camera is closed, cannot switch " + torchState + " torch");
+    private boolean deviceHasFlash(CallbackContext callbackContext) {
+        fragment.hasFlash((boolean result) -> {
+            PluginResult pluginResult1;
+            if (result == true) {
+                pluginResult1 = new PluginResult(PluginResult.Status.OK, true);
+            } else {
+                pluginResult1 = new PluginResult(PluginResult.Status.OK, false);
+            }
+
```

Comment:
```suggestion
            PluginResult pluginResult = new PluginResult(PluginResult.Status.OK, result);
```
can keep the name as pluginResult, it's a local variable